src/options/Options.tsx

```diff
@@ -231,21 +217,15 @@ export default class Options extends Component<Props, State> {
           </div>
         )}
 
-        {/*<Divider horizontal>Domains</Divider>
+        <Divider horizontal>Domains</Divider>
         <Form.Checkbox
           toggle
           label="Cache domain qualities and compare them on the next visit"
           checked={this.state.configuration.cacheDomainQualities}
           onChange={this.toggleCacheDomainQualities}
         />
-        <Form.Checkbox
-          toggle
-          label="Cache domain qualities in Icognito Mode"
-          checked={this.state.configuration.cacheDomainQualitiesIncognito}
-          onChange={this.toggleCacheDomainQualitiesIncognito}
-        />
         <Form.Button content="Remove cache" fluid onClick={this.removeCache} />

```

Comment:
If the question above is answered with yes and cached domains get deleted upon deactivation of the toggle switch then this button should be inactive in case `toggleCacheDomainQualities` is in state deactivated.

Reply:
If you refer to setting the new quality directly on button click in the blocking page, then no.